AEXS/aex-draft_aepps_metadata_format_specification.md

```diff
@@ -0,0 +1,130 @@
+# AEX Z: æpps Metadata Format Specification
+
+```
+AEX: <to be assigned by editors>
+Title: æpps Meta Information Format
+Author: Stoyan Vasilev (@j28), Denis Davidyuk (@davidyuk)
+License: BSD-3-Clause
+Discussions-To: https://forum.aeternity.com/t/aexpansion-proposal-aepps-metadata-format-specification/3987/9
+Status: Draft
+Type: Meta
+Created: 2019-07-09
+```
+
+
+## Simple Summary
+
+The goal of this document is to specify a format for metadata which æpps provide to æpp listing services, decentralized æpp stores, æpp browsers, and wallets.
+
+
+## Abstract
+
+A format which is consistenly used by developers is needed to facilitate the uniform and useful user experience of browsing aepps inside æpps browsers, wallets, and listing services.
+
+
+## Motivation
+
+The motivation behind this aexpansion is to improve the user experience of browsing æpps inside æpps browsers, wallets, and listing services. Our goal is to make it easy for users to recognize the purpose of an æpp/what the æpp allows its users to do and whether users would like to open the æpp and give it access to the account data it requests.
+
+
+## Specification
+
+### Format
+
+**Proposed fields:**
++ Name of aepp
++ Category (select from predermined list)
++ aepp description
++ aepp icon
++ Networks the aepp is available on [mainnet, testnet]
++ Link to aepp (which would contain the metadata specified in this proposal)
++ Age restrictions/rating?
+
+### References
+
+
+Reference visuals for the result of implementing such metadata format are included below.
+
+A service listing dapps:
+
+![](https://i.imgur.com/KZE2nT7.jpg =240x)
+
+Detail view of a dapp metadata rendered inside a dapps browser (Coinbase Wallet):
+
+![](https://i.imgur.com/31evUOm.png =240x)
+
+## Rationale
+
+We are leveraging the webmanifest format and augmenting it to fit the needs of the blockchain context. The premise is that this format can be universal (used by multiple protocols, with only one field difference).
+
+### Other considerations
++ Consider if aeppsmanifest format applies to both web and non-web contexts.
+
++ This proposal does not address whether metadata provided by an aepp is accurate/truthful.
+
+### Future Considerations
+
++ This proposal does not articulate a stance on storage concerns (as aepp metadata can be rendered dynamically or stored locally).
++ This proposal does not address the concern of aepps sharing aepps data between themselves.
+
+
+## Implementation
+
+### Required members
+
+#### `name` member
+Documentation: https://www.w3.org/TR/appmanifest/#name-member
+
+#### `icons` member
+Documentation: https://www.w3.org/TR/appmanifest/#icons-member
+
+#### `aeternity_network_ids` member
```

Comment:
I would name this `network_ids` only if we are aiming for broader blockchain networks adoption, which could use similar structure:

```
network_ids = [
    {   "name"  : "aeternity",
        "ids"   : ["ae_mainnet", "ae_uat"]
    },
    {
        "name"  : "ethereum",
        "ids"   : ["1", "3"]
    },
    {...}
]
```

Reply:
i can see the pros/cons of both approaches. the current implementation is based on the w3c vendor prefix spec/suggestion: https://w3c.github.io/manifest/#proprietary-extensions

since we don't have a meaningful method for voting here. i think i will have to take the risk and making the call here. i'm in favor of the w3c method because it adheres to their guidelines.

Reply:
Sure!